content/realtime/connection.textile

```diff
@@ -393,6 +394,27 @@ h4. Connection recovery constraints
 
 Connection recovery requires that the new client library instance uses credentials that are compatible with those used for the inherited connection; this requires that the same authentication mode is used, with the same key. If token auth was used, the same token is not required, but the token used must have the same @capability@ and <span lang="default">@clientId@</span><span lang="ruby">@client_id@</span><span lang="csharp">@ClientId@</span>. This ensures that the client recovering the connection cannot receive a backlog of messages that its new credentials are not entitled to access. Incompatible credentials will result in an unrecoverable connection error.
 
+h3(#heartbeats). Heartbeats
+
+Heartbeats enable Ably to identify clients that abruptly disconnect from the service, such as where an internet connection drops out or a client changes networks. 
+
+Ably sends a heartbeat to connected clients every 15 seconds. The lack of two consecutive heartbeats will indicate to Ably that a client has disconnected and the "connection state":#connection-states will be updated to @disconnected@. Note that this is only in the case of an abrupt disconnect. Closing a connection, such as by calling "@close()@":#close, is instant.
```

Comment:
This is mixing two different things: the client detecting the connection has dropped, and Ably detecting it. The claim that Ably discovering that a client has disconnected results in that client's connection state being updated is confused: if we're here then we're not talking to the client, so we're not able to communicate anything to it.

```suggestion
Ably sends a heartbeat to connected clients every 15 seconds. If the client goes more than 25 seconds (the heartbeat interval plus a 10s margin of error to allow for network delays) without seeing any server activity from Ably, it assumes that something has gone wrong with the connection and the "connection state":#connection-states will become @disconnected@.

(Ably also uses this mechanism to detect dropped client connections from our end, though the details of this vary by transport).

To be clear, this mechanism is not usually used: it's not needed for the case of a connection deliberately closing (by calling "@close()@":#close ) or being disconnected (by the server or othewise). It is only for when something disrupts communication but does not properly terminate the TCP connection.
```

Reply:
Ah thanks for clarifying that. I've updated the wording based on your suggestion.